nominal/core/dataset.py

```diff
@@ -135,6 +137,104 @@ def add_to_dataset_from_io(
         )
         self._clients.ingest.trigger_file_ingest(self._clients.auth_header, request)
 
+    def get_channel_metadata(
```

Comment:
get_channel is probably sufficient? in general we'll probably keep to all of these resource types being more metadata-y and anything that needs to be downloaded will do so through an additional method & API call on the resource (e.g. the `Attachment.get_contents()` or `Attachment.write()` methods).